partiql-planner/src/main/kotlin/org/partiql/planner/PartiQLHeader.kt

```diff
@@ -590,23 +590,26 @@ public object PartiQLHeader : Header() {
     private fun extract(): List<FunctionSignature.Scalar> = emptyList()
 
     private fun dateAdd(): List<FunctionSignature.Scalar> {
+        val intervals = listOf(INT32, INT64, INT)
```

Comment:
```suggestion
        val intervals = listOf(INT8, INT16, INT32, INT64, INT)
```

Reply:
What if the input is not a literal? Would you still prefer an implicit cast?

Reply:
DateAdd(Day, a_int16_type, b_timestamp_type)

Reply:
Yeah that's fine. We mostly need int32 for our target systems

Reply:
The default was INT which should never be the default. The default integer should always be INT32. Then INT64 is between them, hence why I did those three